### DISCUSSION TITLE
Setting up release branch

### DISCUSSION THREAD

**Discussion by Xeri7** [GitTools/GitVersion#4139] [Q&A] [answered]
Can I get a quick description on how to setup a release branch that will create a non-prerelease version? I have attempted a few things, but we are trying to get a branch structure setup so that our automated deployment can do releases, but right now that requires setting releases explicitly with tags as far as I can tell.

**Comment by HHobeck** [comment, marked as answer]
Have you tried this? Exactly for your use case, we have changed the meaning of Continuous Deployment in the actual version of GitVersion.

> Continuous Deployment
> 
> Continuous Deployment is the process of checking into main and automatically deploying to production.
> 
> For this mode we follow the logic in [this blog post by Xavier Decoster](https://www.xavierdecoster.com/semantic-versioning-auto-incremented-nuget-package-versions) on the issues of incrementing automatically.
> How Continuous Deployment affects GitVersion
> 
> The thing about continuous deployment is that there will be only one version to deploy on production. This means that GitVersion will build the same semantic version for every commit until it has been tagged. For instance:
> 
>     1.2.0
>     1.1.0 (tag: 1.1.0) <-- This is the version which has been deployed on production
>     1.1.0
>     1.1.1

Link[1]: https://gitversion.net/docs/reference/modes/continuous-deployment

**Reply by Xeri7** [reply]
> Have you tried this? Exactly for your use case, we have changed the meaning of Continuous Deployment in the actual version of GitVersion.
> 
> > Continuous Deployment
> > Continuous Deployment is the process of checking into main and automatically deploying to production.
> > For this mode we follow the logic in [this blog post by Xavier Decoster](https://www.xavierdecoster.com/semantic-versioning-auto-incremented-nuget-package-versions) on the issues of incrementing automatically.
> > How Continuous Deployment affects GitVersion
> > The thing about continuous deployment is that there will be only one version to deploy on production. This means that GitVersion will build the same semantic version for every commit until it has been tagged. For instance:
> > ```
> > 1.2.0
> > 1.1.0 (tag: 1.1.0) <-- This is the version which has been deployed on production
> > 1.1.0
> > 1.1.1
> > ```
> 
> Link[1]: https://gitversion.net/docs/reference/modes/continuous-deployment

Hmm, even trying that it still always appends a pre-release number, even on release branches. I may be missing something, but nothing I found there so far lets me have a non-prerelease without manually tagging it

**Reply by Xeri7** [reply]
EDIT: Whoop, somehow i misread the default configs, getting things sorted out now haha

I tried trimming things down to just this, still getting prefixes added on when building on the main/Production branch. Looked like GitHubFlow had Continuous deployment as the default. Theoretically if I understand right, this should be giving no pre-release tags on things in main or release, but neither seems to do so, so not sure what I am missing.
```
workflow: GitHubFlow/v1
tag-prefix: 'version/'
branches:
  main:
    regex: ^Production$
  develop:
    regex: ^Development$
    label: alpha
```
